go/vt/sqlparser/ast.go

```diff
@@ -996,6 +1048,35 @@ func (ts *TableSpec) Format(buf *TrackedBuffer) {
 	buf.Myprintf("\n)%s", strings.Replace(ts.Options, ", ", ",\n  ", -1))
 }
 
+// Format formats the node as a SQL command.
+func (spec *AlterSpec) Format(buf *TrackedBuffer) {
+	switch spec.Action {
+	case AlterAddColumn:
+		buf.Myprintf("add column ")
+		for i, col := range spec.ColumnsAdded {
+			if i == 0 {
+				buf.Myprintf("%v", col)
+			} else {
+				buf.Myprintf(", %v", col)
+			}
+		}
+	case AlterDropColumn:
+		buf.Myprintf("drop column %v", spec.ColumnDropped)
+	case AlterAddIndexOrKey:
+		buf.Myprintf("add %v", spec.IndexAdded)
+	case AlterDropIndexOrKey:
+		buf.Myprintf("drop index %v", *spec.IndexDropped)
+	case AlterAddPartition:
+		buf.Myprintf("add partition %v", spec.PartitionAdded)
+	case AlterDropPartition:
+		buf.Myprintf("drop%v", spec.PartitionsDropped)
```

Comment:
Shouldn't this be `drop partition %v`? Maybe we should add tests cases for the new constructs in `parse_test.go` to make sure that parsing and pretty printing both agree?

Reply:
Annoyingly, it's not and I didn't want to change the way PartitionsDropped worked. We already have test cases for `drop partition`: 

```
	}, {
		input:  "alter table a drop partition p2712",
		output: "alter table a drop partition (p2712)",
	}, {
```

Reply:
This generates a syntax error in mysql:
```
mysql> alter table a drop partition (a,b);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(a,b)' at line 1
mysql> alter table a drop partition a;
ERROR 1505 (HY000): Partition management on a not partitioned table is not possible
mysql> alter table a drop partition a, b;
ERROR 1505 (HY000): Partition management on a not partitioned table is not possible
```

If you want to reuse an existing type, you could use `Exprs` which prints a comma-separated list. Otherwise, I'd recommend creating a new `ColIdents` type so you can define a separate formatting for that member.

Also, I assume partition names are case-insensitive. Otherwise, we should use `TabletIdents`.